types/hapi/definitions/request/request.d.ts

```diff
@@ -0,0 +1,216 @@
+import * as stream from "stream";
+import * as url from "url";
+import * as http from "http";
+import * as Podium from "podium";
+import {PluginsStates, RequestAuth, RequestEvents, RequestInfo, RequestRoute, ResponseObject, Server, Util} from "hapi";
+
+/**
+ * An object containing the values of params, query, and payload before any validation modifications made. Only set when input validation is performed.
+ * For context [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestorig)
+ */
+export interface RequestOrig {
+    params: object;
+    query: object;
+    payload: object;
+}
+
+/**
+ * The request object is created internally for each incoming request. It is not the same object received from the node
+ * HTTP server callback (which is available via [request.raw.req](https://github.com/hapijs/hapi/blob/master/API.md#request.raw)). The request properties change throughout
+ * the request [lifecycle](https://github.com/hapijs/hapi/blob/master/API.md#request-lifecycle).
+ */
+export interface Request extends Podium {
+
+    /**
+     * Application-specific state. Provides a safe place to store application data without potential conflicts with the framework. Should not be used by plugins which should use plugins[name].
+     * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestapp)
+     */
+    app: object;
```

Comment:
Should probably be an interface for user extension.

Reply:
Updated.